### PR TITLE
Add a special case for s390x + py38, needs fix in NumPy 1.17.5

### PR DESCRIPTION
**Pull request by rgommers**
Closes gh-29

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -40,6 +40,9 @@ install_requires =
     numpy==1.21.0; python_version=='3.8' and platform_machine=='arm64' and platform_system=='Darwin'
     numpy==1.21.0; python_version=='3.9' and platform_machine=='arm64' and platform_system=='Darwin'
 
+    # Python 3.8 on s390x requires at least 1.17.5, see gh-29
+    numpy==1.17.5; python_version=='3.8' and platform_system=='s390x' and platform_python_implementation != 'PyPy'
```
Comment (pitrou) [inline setup.cfg:44]:
```suggestion
    numpy==1.17.5; python_version=='3.8' and platform_machine=='s390x' and platform_python_implementation != 'PyPy'
```

**Comment by pitrou** [comment]
Checked on Travis-CI:
https://app.travis-ci.com/github/pitrou/arrow/jobs/551378494#L234

```
$ python -c "import platform, sys; print(sys.platform); print(platform.machine())"
linux2
s390x
```

**Comment by rgommers** [comment]
Great, thanks. Okay I'll merge this and will cut a new release of this package, then it should resolve the problem in the Arrow PR where you hit this problem.

**Comment by pitrou** [comment]
Thank you very much :-)

**Pull request merged**